### PR TITLE
GH-8: Shared KafkaProducer

### PR DESCRIPTION
**Pull request by garyrussell**
Resovles #8

### CODE REVIEW

File: spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java [hunk 1]
```diff
@@ -16,26 +16,128 @@
...
 	public DefaultKafkaProducerFactory(Map<String, Object> configs) {
 		this.configs = new HashMap<>(configs);
 	}
 
 	@Override
+	public void destroy() throws Exception {
+		Producer<K, V> producer = this.producer;
+		this.producer = null;
+		if (producer != null) {
+			producer.close();
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java]: Is it expected?Looking into the `this.producer = new CloseSafeProducer<K, V>()`, and its `no-op` `close()` impl, I'd say that this is a dead code and we don't need this `start()`, `stop()` and `destroy()` "hell".

Or have you meant something like `producer.delegate.close();` ?

Comment (garyrussell) [inline spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java]: good catch!

**Comment by garyrussell** [comment]
Pushed.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/c8b723fa072d0cc10cee46d825b3bc206dd00c64

**Pull request closed without merging**